websocket/websocket-server.go

```diff
@@ -0,0 +1,335 @@
+package websocket
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"github.com/gorilla/websocket"
+
+	runtime "github.com/Hammad887/chat-app"
+	domain "github.com/Hammad887/chat-app/models"
+)
+
+// ctx is a variable of type context.Context, which is used to manage request-scoped values, deadlines, and cancellations.
+var ctx context.Context
+
+// upgrader use for upgrading http connection to websocket connection
+var upgrader = websocket.Upgrader{
+	// ReadBufferSize specifies the size of the buffer used for reading from the WebSocket connection.
+	ReadBufferSize: 1024,
+
+	// WriteBufferSize specifies the size of the buffer used for writing to the WebSocket connection.
+	WriteBufferSize: 1024,
+
+	// EnableCompression allows for message compression when set to true.
+	EnableCompression: true,
+
+	// CheckOrigin is a function that checks the origin of the WebSocket request to prevent cross-origin attacks.
+	// In this case, it's set to always return true, meaning any origin is accepted.
+	CheckOrigin: func(r *http.Request) bool {
+		return true
+	},
+}
+
+// RunWebsocketServer initializes and runs the WebSocket server.
+func RunWebsocketServer(rt *runtime.Runtime) {
+
+	router := mux.NewRouter()
+
+	// Set up CORS handling
+	headers := handlers.AllowedHeaders([]string{"*"})
+	origins := handlers.AllowedOrigins([]string{"*"}) // Allow requests from any origin
+
+	// Add CORS middleware
+	router.Use(handlers.CORS(headers, origins))
+
+	router.HandleFunc("/ws", func(w http.ResponseWriter, r *http.Request) {
+		ClientSocket(w, r, rt)
+	})
+	fmt.Println("Server is running on port 8085")
+
+	// Start your server
+	go func() {
+		err := http.ListenAndServe(":8085", router)
+		if err != nil {
+			log.Println(err)
+		}
+	}()
+}
+
+// ClientSocket handles incoming WebSocket connections from clients.
+func ClientSocket(w http.ResponseWriter, r *http.Request, rt *runtime.Runtime) {
+
+	Conn, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		log.Print("upgrade:", err)
```

Comment:
handle error correctly, it should pass to the  handler and return error message

Reply:
Done